File: bookmarker-java/src/main/java/com/sivalabs/bookmarker/domain/entity/Tag.java
package com.sivalabs.bookmarker.domain.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotEmpty;
import java.util.Set;

@Entity
@Table(name = "tags")
@Setter
@Getter
public class Tag extends BaseEntity {

    @Id
    @SequenceGenerator(name = "tag_id_generator", sequenceName = "tag_id_seq", allocationSize = 1)
    @GeneratedValue(generator = "tag_id_generator")
    private Long id;

    @Column(nullable = false, unique = true)
    @NotEmpty()
    private String name;

    @JsonIgnore
    @ManyToMany(mappedBy = "tags")
    private Set<Bookmark> bookmarks;
}

File: backend/src/main/kotlin/com/sivalabs/bookmarker/bookmarks/entity/Tag.kt
package com.sivalabs.bookmarker.bookmarks.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.ManyToMany
import javax.persistence.SequenceGenerator
import javax.persistence.Table

@Entity
@Table(name = "tags")
class Tag {

    @Id
    @SequenceGenerator(name = "tag_id_generator", sequenceName = "tag_id_seq", allocationSize = 1)
    @GeneratedValue(generator = "tag_id_generator")
    var id: Long = 0

    @Column(nullable = false, unique = true)
    var name: String = ""

    @JsonIgnore
    @ManyToMany(mappedBy = "tags")
    var bookmarks: MutableList<Bookmark> = mutableListOf()

    override fun toString(): String {
        return "Tag(id=$id, name='$name')"
    }
}
